### DISCUSSION TITLE
Cloudsmith Debian repository mirrors have moved (months ago)

### DISCUSSION THREAD

**Discussion by jogoossens** [rabbitmq/rabbitmq-server#11830] [Other] [answered]
### Describe the bug

The rabbitmq repo on ppa.novemberain.com seems down all the sudden (not found). I see ppa1.novemberain.com still works, any reason the ppa stopped working?

### Reproduction steps

This has worked for very long time without any problems, suddenly the paths / binaries are not found when using ppa.novemberain.com.

cat rabbitmq.sources
```
X-Repolib-Name: RabbitMQ
Types: deb deb-src
URIs: https://ppa.novemberain.com/rabbitmq/rabbitmq-server/deb/debian
Suites: bookworm
Components: main
Signed-By: /etc/apt/keyrings/rabbitmq.asc
```

```
# apt-get update
Hit:1 http://deb.debian.org/debian bookworm InRelease
Hit:2 http://deb.debian.org/debian bookworm-updates InRelease
Hit:3 http://security.debian.org bookworm-security InRelease
Hit:4 http://deb.debian.org/debian bookworm-backports InRelease
Hit:5 https://artifacts.elastic.co/packages/8.x/apt stable InRelease
Hit:6 https://deb.nodesource.com/node_20.x nodistro InRelease
Hit:7 https://repo.percona.com/ppg-16/apt bookworm InRelease
Hit:8 https://repo.percona.com/tools/apt bookworm InRelease
Hit:9 https://packages.icinga.com/debian icinga-bookworm InRelease
Hit:10 https://packages.sury.org/php bookworm InRelease
Ign:11 https://ppa.novemberain.com/rabbitmq/rabbitmq-erlang/deb/debian bookworm InRelease
Hit:12 https://download.bareos.com/bareos/release/23/Debian_12  InRelease
Ign:13 https://ppa.novemberain.com/rabbitmq/rabbitmq-server/deb/debian bookworm InRelease
Err:14 https://ppa.novemberain.com/rabbitmq/rabbitmq-erlang/deb/debian bookworm Release
  404  Not Found [IP: 2606:4700:3034::6815:1b07 443]
Err:15 https://ppa.novemberain.com/rabbitmq/rabbitmq-server/deb/debian bookworm Release
  404  Not Found [IP: 2606:4700:3034::6815:1b07 443]
Hit:16 http://packages.blackfire.io/debian any InRelease
Reading package lists... Done
E: The repository 'https://ppa.novemberain.com/rabbitmq/rabbitmq-erlang/deb/debian bookworm Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'https://ppa.novemberain.com/rabbitmq/rabbitmq-server/deb/debian bookworm Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.


```

### Expected behavior

Being able to find and install the software as it worked before.

**Comment by michaelklishin** [comment, marked as answer]
@jogoossens `ppa.novemberain.com` was replaced with `ppa1.novemberain.com` and `ppa2.novemberain.com` many months ago, as [can be seen in the Debian installation guide](https://www.rabbitmq.com/docs/install-debian#apt-quick-start-cloudsmith).

During an upgrade last week the original `ppa.novemberain.com` host was permanently removed. Again, after close to a year of a "transition period."

The new mirrors have served 39M requests in the last four weeks, and serve a few requests a second right now according to the logs.

**Comment by michaelklishin** [comment]
The first course of action when a recommended installation method fails should not be "file an issue on GitHub" and not even "start a discussion on GitHub", it should be "consult the documentation, maybe something has changed."

There will be other package infrastructure changes later this year, and while they won't affect the mirrors in question, they will affect some users who have set up their `/etc/apt/sources.list.d` files years ago and haven't looked at them since. In many of these migrations our team does not have a choice: services such as Bintray go away, our team is forced one infrastructure provide over the other by our corporate overlords, and so on.

**Comment by jogoossens** [comment]
Ok thx a lot, now it's clear it's on purpose, I was under the (wrong) impression the ppa one would always point to a working one , sorry :)

**Reply by michaelklishin** [reply]
We could re-introduce it but the problem is, the way apt and Cloudflare work makes a "load balancing" domain name such as `ppa.rabbitmq.com` mostly useless. You must list `ppa1.novemberain.com` (future `ppa1.rabbitmq.com`), `ppa2.novemberain.com`, and so on, and the docs do that. Otherwise you do not really get much redundancy in terms of `apt`-reported errors.

For example, should `ppa1.novemberain.com` miss a file or something like that, when a request is routed from `ppa.novemberain.com` to that instance, you'd still get an `apt` error. If you also list `ppa2.novemberain.com` explicitly in the `sources.d` file, `apt` will try that repo, find the package (or a metadata file) there and complete the operation successfully.
